week03/assignment03_test.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestRate(t *testing.T) {
+
+	var tests = []struct {
+		a    Movie
+		want error
+	}{
+		{Movie{Name: "Gladiator", Length: 155, plays: 4}, nil},
+		{Movie{Name: "Léon: The Professional", Length: 110}, fmt.Errorf("can't review a movie without watching it first")},
+	}
+
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
+			got := tt.a.Rate(50.2)
+			if got != tt.want {
+				t.Errorf("no movies to play: %s", got)
+			}
+		})
+
+	}
+}
+
+func TestPlay(t *testing.T) {
+	var tests = []struct {
+		a    Movie
+		b    int
+		want int
+	}{
+		{Movie{Name: "Gladiator", Length: 155}, 5, 5},
+		{Movie{Name: "Léon: The Professional", Length: 110}, 5, 0},
+	}
+
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
+			tt.a.Play(tt.b)
+			if tt.a.viewers != tt.want {
+				t.Errorf("the %s movie views %d, expected: %d.", tt.a.Name, tt.a.viewers, tt.want)
+			}
+			if tt.a.plays != tt.want {
+				t.Errorf("the %s movie plays %d, expected: %d.", tt.a.Name, tt.a.plays, tt.want)
+			}
+		})
+
+	}
+}
+
+func TestViewers(t *testing.T) {
+	var tests = []struct {
+		a Movie
+	}{
+		{Movie{Name: "Gladiator", Length: 155, viewers: -1}},
+		{Movie{Name: "Léon: The Professional", Length: 110, viewers: 4}},
+	}
+
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
+			got := tt.a.Viewers()
+			if reflect.TypeOf(got) != reflect.TypeOf(0) || got < 0 {
```

Comment:
There is no need for use of reflection here. `got` is a concrete type, `int`. All this code does is confirm what the compiler is already doing. 
```suggestion
			if got < 0 {
```

Reply:
ok